### PR TITLE
[Sprint 5] Feature mainpage user book detail

### PR DESCRIPTION
**Pull request by tranminhtri911**
- https://dev.framgia.com/redmine/issues/138123
- Feature mainpage user book detail
- Effect : None

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -58,6 +58,12 @@
     <string name="choose_category">Choose Category</string>
     <string name="this_book_does_not_have_the_owner">This book does not have the owner !</string>
     <string name="you_already_own_this_book">You already own this book !</string>
+    <string name="view_details">View details</string>
+    <string name="user_review">User Review</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: WARNING: [UnusedResources] The resource `R.string.user_review` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/values-vi/strings.xml [hunk 2]
```diff
@@ -47,6 +47,12 @@
     <string name="choose_category">Chọn thể loại</string>
     <string name="this_book_does_not_have_the_owner">Cuốn sách này không có chủ sở hữu !</string>
     <string name="you_already_own_this_book">Bạn đã sở hữu cuốn sách này !</string>
+    <string name="view_details">Xem chi tiết</string>
+    <string name="user_review">Đánh giá</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values-vi/strings.xml]: WARNING: [UnusedResources] The resource `R.string.user_review` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -58,6 +58,12 @@
...
+    <string name="user_waiting">User Waiting</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: WARNING: [UnusedResources] The resource `R.string.user_waiting` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/values-vi/strings.xml [hunk 4]
```diff
@@ -47,6 +47,12 @@
...
+    <string name="user_waiting">Đang đợi</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values-vi/strings.xml]: WARNING: [UnusedResources] The resource `R.string.user_waiting` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -58,6 +58,12 @@
...
+    <string name="user_reading">User Reading</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: WARNING: [UnusedResources] The resource `R.string.user_reading` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/values-vi/strings.xml [hunk 6]
```diff
@@ -47,6 +47,12 @@
...
+    <string name="user_reading">Đang đọc</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values-vi/strings.xml]: WARNING: [UnusedResources] The resource `R.string.user_reading` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -58,6 +58,12 @@
...
+    <string name="user_returning">User Returning</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: WARNING: [UnusedResources] The resource `R.string.user_returning` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/values-vi/strings.xml [hunk 8]
```diff
@@ -47,6 +47,12 @@
...
+    <string name="user_returning">Đang trả lại</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values-vi/strings.xml]: WARNING: [UnusedResources] The resource `R.string.user_returning` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -58,6 +58,12 @@
...
+    <string name="user_returned">User Returned</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: WARNING: [UnusedResources] The resource `R.string.user_returned` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/values-vi/strings.xml [hunk 10]
```diff
@@ -47,6 +47,12 @@
...
+    <string name="user_returned">Đã trả lại</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values-vi/strings.xml]: WARNING: [UnusedResources] The resource `R.string.user_returned` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/layout/activity_bookdetail.xml [hunk 11]
```diff
@@ -333,12 +333,31 @@
...
                         android:layout_marginStart="@dimen/dp_20"
                         android:layout_marginTop="@dimen/dp_10"
-                        android:text="@string/activities"
-                        android:textAllCaps="true"
-                        android:textColor="@color/color_dark_gray"
-                        android:textSize="@dimen/sp_17"
-                        android:textStyle="bold"
-                        />
+                        >
+                        <TextView
```
Comment (trainer-android-fr) [inline app/src/main/res/layout/activity_bookdetail.xml]: WARNING: [RelativeOverlap] `TextView-1` can overlap `TextView-3` if @string/activities, @string/view_details grow due to localized text expansion
 If relative layout has text or button items aligned to left and right sides they can overlap each other due to localized text expansion unless they have mutual constraints like `toEndOf`/`toStartOf`.

File: app/src/main/java/com/framgia/fbook/screen/bookdetail/BookDetailActivity.kt [hunk 12]
```diff
@@ -81,6 +84,14 @@ open class BookDetailActivity : BaseActivity(), BookDetailContract.ViewModel, It
     super.onStop()
   }
 
+  override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+    super.onActivityResult(requestCode, resultCode, data)
+    if (resultCode == Activity.RESULT_OK && requestCode == Constant.RequestCode.BOOK_DETAIL_REQUEST) {
+      presenter.getBookDetail(mBookId)
```
Comment (HyperionsThs) [inline app/src/main/java/com/framgia/fbook/screen/bookdetail/BookDetailActivity.kt]: change to presenter -> mPresenter =))

File: app/src/main/java/com/framgia/fbook/screen/userinbookdetail/UserInBookDetailActivity.kt [hunk 13]
```diff
@@ -1,20 +1,38 @@
...
 import javax.inject.Inject
 
+
 /**
  * UserInBookDetail Screen.
  */
 open class UserInBookDetailActivity : BaseActivity(), UserInBookDetailContract.ViewModel {
 
   @Inject
   internal lateinit var presenter: UserInBookDetailContract.Presenter
```
Comment (HyperionsThs) [inline app/src/main/java/com/framgia/fbook/screen/userinbookdetail/UserInBookDetailActivity.kt]: change to presenter -> mPresenter

File: app/src/main/res/layout/activity_bookdetail.xml [hunk 14]
```diff
@@ -333,12 +333,31 @@
...
+                            android:layout_height="wrap_content"
+                            android:layout_alignParentLeft="true"
+                            android:layout_alignParentStart="true"
+                            android:text="@string/activities"
+                            android:textAllCaps="true"
+                            android:textColor="@color/color_dark_gray"
+                            android:textSize="@dimen/sp_17"
+                            android:textStyle="bold"
+                            />
+                        <TextView
```
Comment (trainer-android-fr) [inline app/src/main/res/layout/activity_bookdetail.xml]: WARNING: [RelativeOverlap] `TextView-3` can overlap `TextView-1` if @string/activities, @string/view_details grow due to localized text expansion
 If relative layout has text or button items aligned to left and right sides they can overlap each other due to localized text expansion unless they have mutual constraints like `toEndOf`/`toStartOf`.

**Pull request merged**